packages/typescript/src/internal/devserver/ts_devserver.bzl

```diff
@@ -86,8 +87,14 @@ def _ts_devserver(ctx):
         for f in script_files
     ]))
 
+    # With cross-platform RBE for OSX & Windows ctx.executable.devserver will be linux as --cpu and
+    # --host_cpu must be overridden to k8. However, we still want to be able to run the devserver on the host
+    # machine so we need to include the host devserver binary, which is ctx.executable.devserver_host, in the
+    # runfiles. For non-RBE and for RBE with a linux host, ctx.executable.devserver & ctx.executable.devserver_host
+    # will be the same binary.
     devserver_runfiles = [
         ctx.executable.devserver,
+        ctx.executable.devserver_host,
```

Comment:
feels really gross to upload both binaries to RBE just because it can't select the right one. This is a limitation of cross-platform RBE being broken? Still feels like a hacky workaround

Reply:
I guess you mean this one: https://github.com/bazelbuild/rules_nodejs/blob/dcdc98b093edf2a3ccd6962d8b4f15aa28ba6fa8/internal/common/windows_utils.bzl#L63

It's basically a hack to make the code simpler, but it should work.

Reply:
Hmm we also need a way to distinguish mac from linux, do you know of any trick to do that?

Reply:
Hmm, maybe you can consider this:

Adding the following to .bazelrc

```
--enable_platform_specific_config
build:windows --action_env=NODE_PLATFORM=windows
build:macos --action_env=NODE_PLATFORM=macos
build:linux --action_env=NODE_PLATFORM=linux
```

Then you can check the value of `ctx.configuration.default_shell_env['NODE_PLATFORM']`

Doc for [--enable_platform_specific_config](https://docs.bazel.build/versions/master/command-line-reference.html#flag--enable_platform_specific_config)

Reply:
that's clever but we can't control users' .bazelrc files so this will get us in trouble